### PR TITLE
5.next: route action check in `bin/cake routes`

### PR DESCRIPTION
**Pull request by LordSimal**
image

This adds another column to the `bin/cake routes` command to indicate, if the configured controller+action combination is actually present or not.

e.g.
```
$builder->connect('/check-timesheet', ['controller' => 'Tools', 'action' => 'checkTimesheet']);
```

would required, that the `ToolsController` has a `checkTimesheet` action (which in my case it didn't)

This command should help people get rid of unnecessary routes

As always - my naming schema may be not up2standard so please feel free to provide better names 😉

### CODE REVIEW

File: tests/TestCase/Command/RoutesCommandTest.php [hunk 1]
```diff
@@ -73,6 +73,7 @@ public function testRouteList(): void
             '<info>Prefix</info>',
             '<info>Controller</info>',
             '<info>Action</info>',
+            '<info>Valid Action</info>',
```
Comment (dereuromark) [inline tests/TestCase/Command/RoutesCommandTest.php]: Could also be
```suggestion
            '<info>Existing</info>',
```

**Comment by LordSimal** [comment]
I just realized this creates false positives in the case of 
```
$builder->connect('/mitglieder-projekte/{action}/*', ['controller' => 'StaffMembersProjects']);
```
since the default action at least according to the router is always `index` which my `StaffMembersProjectsController` doesn't have...

**Comment by ADmad** [comment]
This will create false positives and potentially confuse users.

Also there are use cases where the route is set just to identify the URL. For e.g. my social auth plugin sets routes to non-existent controller and the response is return by it's middleware after checking the controller/action name.

**Comment by LordSimal** [comment]
I moved the additional column to the verbose only mode.

@ADmad do you mean this should rather be a separate command in its own plugin rather than in the core?

**Comment by ADmad** [comment]
> do you mean this should rather be a separate command in its own plugin rather than in the core?

Yeah, I am not too keen on having features giving potentially false indications in the core.

**Pull request closed without merging**